lib/stat_tracker.rb

```diff
@@ -23,6 +23,40 @@ def initialize(files)
     end
   end
 
+  def most_accurate_team(season)
+  # name of team with the best ratio of shots to goals for the season 
+  # 1. pull out all game_id for a season in @games as an array
+  # 2. iterate over @game_teams using our created array of game_id as reference
+    # make hash of team_id as keys and set values to [goals + itself, shots + itself]
+  # 3. use transform_values on hash to give goals / shots .round(4)
+  # 4. Find the max value and subsequent key and return the team_name with team_list
+
+  # 1. array of all game_id from a season
+  all_season_game_id = @games.map do |game|
+    game.game_id if game.season == season
+  end.compact
+
+  # 2. hash of team_id as keys and [goals, values] as values
+  team_id_goals_shots = @game_teams.each_with_object(Hash.new([0,0])) do |game, hash|
+    # if game.game_id exists in our all_season_game_id array, then create key-value pair
+    if all_season_game_id.include?(game.game_id)
+      hash[game.team_id] = [game.goals + hash[game.team_id][0], game.shots + hash[game.team_id][1]]
```

Comment:
incrementing the goals and shots? This is a bit brain bendy but I think I understand what's happening here. Might have to chat this one over for further understanding, but seems to be working!